### ISSUE TITLE
[Task] Accessibility

### ISSUE THREAD

**Issue by larevolucia** [larevolucia/chaptr#87]
### Description
Ensure site has accessibility features, such as aria-labels, contrast ratio, and keyboard navigation.

### Checklist (if applicable)
- [ ] Test and adjust accessibility feeatures according to https://www.accessibilitychecker.org/ , https://wave.webaim.org/ and lighthouse
- [ ] Test and adjust keyboard navigation 
- [ ] Test and adjust "skip to main" in all pages

**Comment by larevolucia** [comment]
Desktop:

<img width="707" height="371" alt="Image" src="https://github.com/user-attachments/assets/a2bc45cc-46e9-4c64-a924-e1729d7221b3" />

Mobile:

<img width="719" height="413" alt="Image" src="https://github.com/user-attachments/assets/339dd89d-cba6-4968-b698-7339fdc4f308" />

**Comment by larevolucia** [comment]
Rating component needs fixing for keyboard navigation. Currently a user cannot rate a book using keyboard navigation.

**Comment by larevolucia** [comment]
skip to main works, but part of the content is hidden by nav bar

**Comment by larevolucia** [comment]
Detail Page:

<img width="892" height="195" alt="Image" src="https://github.com/user-attachments/assets/570cd87a-7af8-4621-898a-c0f44466dc8b" />

Search Page: 

<img width="896" height="199" alt="Image" src="https://github.com/user-attachments/assets/2ee5234a-100e-4ec8-8f54-2d7f468b64bb" />

Wave Aim passed for both, only issue was contrast for ellipsis on search results pagination.

**Comment by larevolucia** [comment]
Lighthouse

Home:

Deprecated APIs will eventually be removed from the browser. Learn more about deprecated APIs. Deprecation / Warning Source Heroku other 1st party Found an <h1> tag within an <article>, <aside>, <nav>, or <section> which does not have a specified font-size. The size of this heading text will be changing in this browser in the near future. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#specifying_a_uniform_font_size_for_h1 for more information.

Fix: set explicit sizes on CSS

**Comment by larevolucia** [comment]
About http to https:

https://forum.djangoproject.com/t/make-http-redirect-to-https/12678/2

https://forum.djangoproject.com/t/make-http-redirect-to-https/12678

https://www.pullrequest.com/blog/implementing-and-auditing-https-in-django-best-practices/#:~:text=Use%20Django%20Security%20Middleware&text=SECURE_SSL_REDIRECT%3A%20When%20set%20to%20True,site%20over%20a%20secure%20connection.

**Comment by larevolucia** [comment]
`Does not use HTTPS 7 insecure requests found All sites should be protected with HTTPS, even ones that don't handle sensitive data. This includes avoiding mixed content, where some resources are loaded over HTTP despite the initial request being served over HTTPS. HTTPS prevents intruders from tampering with or passively listening in on the communications between your app and your users, and is a prerequisite for HTTP/2 and many new web platform APIs. Learn more about HTTPS. Insecure URL Request Resolution /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS /books/content?id=…(books.google.com) Automatically upgraded to HTTPS Issues were logged in the Issues panel in Chrome Devtools Issues logged to the Issues panel in Chrome Devtools indicate unresolved problems. They can come from network request failures, insufficient security controls, and other browser concerns. Open up the Issues panel in Chrome DevTools for more details on each issue. Issue type Mixed content /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) /books/content?id=…(books.google.com) Incorrect use of <label for=FORM_ELEMENT> The label's for attribute refers to a form field by its name, not its id. This might prevent the browser from correctly autofilling the form and accessibility tools from working correctly. To fix this issue, refer to form fields by their id attribute. 2 resources Label Label Learn more: HTML Standard`

To fix http best practices warning from Lighthouse, I added a new utils funciont and normalized on the functions in the services.py

def ensure_https(url: Optional[str]) -> Optional[str]:
    """Force http:// → https:// once, leave None/https as-is."""
    if not url:
        return url
    return re.sub(r"^http:", "https:", url, count=1)

**Comment by larevolucia** [comment]
Search Results:

mobile
<img width="731" height="380" alt="Image" src="https://github.com/user-attachments/assets/c9579d38-0624-4e6f-87b0-14097a967e39" />
desktop
<img width="731" height="439" alt="Image" src="https://github.com/user-attachments/assets/e297c305-aa92-4c63-a4d0-daf16664f11a" />

**Comment by larevolucia** [comment]
Largest Contentful Paint image was lazily loaded Above-the-fold images that are lazily loaded render later in the page lifecycle, which can delay the largest contentful paint. Learn more about optimal lazy loading.LCP Element img.mb-2.img-card-search Largest Contentful Paint element 4,090 ms This is the largest contentful element painted within the viewport. 
Learn more about the Largest Contentful Paint element

**Comment by larevolucia** [comment]
When running Lighthouse on a non-incognito tab we get performance issues due to Third Party Cookies. 

<img width="557" height="415" alt="Image" src="https://github.com/user-attachments/assets/f85a0ab1-1098-4e32-ad3e-9e0e5c093ddb" />

https://privacysandbox.google.com/cookies/prepare/overview
https://developer.mozilla.org/en-US/docs/Web/Privacy/Guides/Third-party_cookiess
https://privacysandbox.google.com/cookies/prepare/audit-cookies?utm_source=chatgpt.com

**Comment by larevolucia** [comment]
fixed third-party cookies issue with cover_proxy

<img width="551" height="401" alt="Image" src="https://github.com/user-attachments/assets/1fb33855-0dc2-4b11-b239-7864aa96d034" />

<img width="418" height="65" alt="Image" src="https://github.com/user-attachments/assets/bdc9f9ad-320b-40c6-97ad-2b86c0bf9770" />
